admin/tabs/general/roles/RoleModel.ts

```diff
@@ -207,6 +221,32 @@ export class RoleModel extends HoistModel {
         });
     }
 
+    private processRolesForTreeGrid(
+        roles: Omit<HoistRole, 'users' | 'directoryGroups' | 'roles'>[]
+    ) {
+        const topList = [];
+        roles.forEach(role => {
+            const categories = role.category ? role.category.split('\\') : ['Uncategorized'];
+
+            let list = topList,
+                id = '';
+            categories.forEach(category => {
+                let currCat = list.find(it => it.name === category);
```

Comment:
Should you also be checking that `isGroupRow === true`? What if you had a role with the same name as a category?